packages/kit/src/runtime/server/respond.js

```diff
@@ -457,6 +457,10 @@ export async function respond(request, options, manifest, state) {
 				return response;
 			}
 
+			if (state.error && state.depth > 0) {
+				return await fetch(request);
```

Comment:
I think this could cause an infinite chain of requests.

Eg:
A request comes for `kit.svelte.dev/non-existent`, which is a 404 so it tries to render the error page, which makes a request to `/nav.json` but maybe that had a typo in the path and then that response also tries to do the same... and so on?

Reply:
I borked it

![image](https://github.com/sveltejs/kit/assets/50981692/dc5b89f3-58f2-4fac-90e7-e4700373e570)

Reply:
Gah there's no easy solution, is there? 😓 How could we catch this case of "error from within error state", which should probably reroute to the fallback `error.html`?

Reply:
That'd kick in if the state was saved, but with the fetch it ends up being a completely new request that the server doesn't know it's related to the previous one in any way.

Reply:
Hmm, maybe a custom header to identify the request? Or do we generally avoid adding extra headers?

Reply:
I've added a header to the fetch request so the server can identify when we're coming from an error state. It fallbacks to the `error.html` page instead.